data-prepper-core/src/test/java/org/opensearch/dataprepper/core/pipeline/buffer/ZeroBufferTests.java

```diff
@@ -211,6 +211,19 @@ public void testReadFromEmptyBufferReturnsNoRecords() {
             assertTrue(readRecordsMap.getKey().isEmpty());
             verify(pipelineRunner, never()).runAllProcessorsAndPublishToSinks();
         }
+
+        @Test
+        public void testReadFromEmptyBufferCallsThreadSleep() {
+            ZeroBuffer<Record<String>> zeroBuffer = createObjectUnderTest();
+            
+            long startTime = System.currentTimeMillis();
```

Comment:
Our build are taking a long time and tests with sleeps contribute to this. I think using Mockito's mockStatic on Thread.sleep() is better here. I think some other tests do this.

Reply:
Updated the test to test Thread.sleep() call indirectly through Thread.currentThread().interrupt() instead.